GWXtreme/eos_prior.py

```diff
@@ -0,0 +1,338 @@
+# Copyright (C) 2017 Daniel Wysocki
+# Assimilated from Wysocki D., O'Shaughnessy R., Bayesian Parametric Population Models, 
+# 2017–, bayesian-parametric-population-models.readthedocs.io
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+import typing
+import numpy
+import scipy.stats
+import h5py
+
+
+largest_ns_mass = 1.97
+
+def eos_p_of_rho(rho,eos):
+    import lalsimulation
+    FAM=lalsimulation.CreateSimNeutronStarFamily(eos)
+    p_max_i = min(6e35, lalsimulation.SimNeutronStarEOSMaxPressure(eos))
+    log10_p_grid = numpy.linspace(*numpy.log10([5e31, p_max_i]), 128)
+    p_grid = numpy.power(10.0, log10_p_grid)
+    rho_grid = numpy.empty_like(p_grid)
+    for j, p in numpy.ndenumerate(p_grid):
+                h = lalsimulation.SimNeutronStarEOSPseudoEnthalpyOfPressure(p, eos)
+                rho_grid[j] = (
+                    lalsimulation.SimNeutronStarEOSRestMassDensityOfPseudoEnthalpy(
+                        h, eos,
+                    )
+                )
+
+
+    log10_rho_grid = numpy.log10(rho_grid)
+    log10_p_out = numpy.interp(
+                numpy.log10(rho), log10_rho_grid, log10_p_grid,
+                left=numpy.NINF, right=numpy.NINF,
+            )
+    return log10_p_out
+
+
+def spectral_eos(eos_parameters: tuple) -> typing.Any:
+    import lalsimulation
```

Comment:
Just import `lalsimulation` globally as is done in this package. Also please add docstring.